src/main/java/com/cloudant/client/api/DesignDocumentManager.java

```diff
@@ -0,0 +1,213 @@
+     */
+    public DesignDocument get(String id) {
+        return db.find(DesignDocument.class, id);
+    }
+
+    /**
+     * Gets a design document using the id and rev from the database.
+     *
+     * @param id  The document id
+     * @param rev The document revision
+     * @return {@link DesignDocument}
+     */
+    public DesignDocument get(String id, String rev) {
+        assertNotEmpty(id, "id");
+        assertNotEmpty(id, "rev");
+        return db.find(DesignDocument.class, id, rev);
+    }
+
+    /**
+     * Removes a design document from the database.
+     *
+     * @param id The document id
+     * @return {@link DesignDocument}
+     */
+    public Response remove(String id) {
+        assertNotEmpty(id, "id");
+        DesignDocument find = db.find(DesignDocument.class, id);
+        return db.remove(find.getId(), find.getRevision());
+    }
+
+    /**
+     * Removes a design document using the id and rev from the database.
+     *
+     * @param id  The document id
+     * @param rev The document revision
+     * @return {@link DesignDocument}
+     */
+    public Response remove(String id, String rev) {
+        assertNotEmpty(id, "id");
+        assertNotEmpty(id, "rev");
+        return db.remove(id, rev);
+
+    }
+
+    /**
+     * Removes a design document using DesignDocument object from the database.
+     *
+     * @param designDocument the design document object to be removed
+     * @return {@link DesignDocument}
+     */
+    public Response remove(DesignDocument designDocument) {
+        assertNotEmpty(designDocument, "DesignDocument");
+        return db.remove(designDocument);
+    }
+
+    /**
+     * Deserialize a directory of javascript design documents to a DesignDocument object.
+     *
+     * @param directory the directory containing javascript files
+     * @return {@link DesignDocument}
+     */
+    public static List<DesignDocument> fromDirectory(File directory) {
+        List<DesignDocument> designDocuments = new ArrayList<DesignDocument>();
+        if (directory.isDirectory()) {
+            Collection<File> files = FileUtils.listFiles(directory, null, true);
+            for (File designDocFile : files) {
+                designDocuments.add(fromFile(designDocFile));
+            }
+        } else {
+            designDocuments.add(fromFile(directory));
+        }
+        return designDocuments;
+    }
+
+    /**
+     * Deserialize a javascript design document to a DesignDocument object.
+     *
+     * @param file the design document javascript file
+     * @return {@link DesignDocument}
+     */
+    public static DesignDocument fromFile(File file) {
+        assertNotEmpty(file, "Design js file");
+        DesignDocument designDocument;
+        Gson gson = new Gson();
+        //Deserialize JS file contents into DesignDocument object
+        try {
+            designDocument = gson.fromJson(new FileReader(file), DesignDocument.class);
+        } catch (FileNotFoundException e) {

```

Comment:
Can we just throw this rather than wrapping it in a `RuntimeException`? I think it is more useful to the end user to know what kind of errors they might expect.